test/jdk/javax/swing/JFileChooser/4150029/bug4150029.java

```diff
@@ -21,73 +21,119 @@
  * questions.
  */
 
-/*
-   bug 4150029 8006087
-   summary BackSpace keyboard button does not lead to parent directory
-   author Oleg Mokhovikov
-*/
+import java.awt.BorderLayout;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
 
 import jdk.test.lib.Platform;
 
-import javax.swing.*;
-import java.io.File;
-import java.io.IOException;
+/*
+ * @test
+ * @bug 4150029 8006087
+ * @key headful
+ * @summary BackSpace keyboard button does not lead to parent directory
+ * @library /test/lib
+ * @build jdk.test.lib.Platform
+ * @run main bug4150029
+ */
 
-public class bug4150029 extends JApplet {
-    private boolean res;
+public class bug4150029 {
+    private static JFrame frame;
+    private static JFileChooser fileChooser;
+    private static Robot robot;
+    private static File testDir;
+    private static File subDir;
```

Comment:
`testDir` can be declared as local variable.
Probably `subDir` also can be moved as local variable provided you create JFileChooser instance with default directory and then set current directory inside main method after createAndShowUI method call.

something like this..

`SwingUtilities.invokeAndWait(() -> {
                createAndShowUI();
                fileChooser.setCurrentDirectory(subDir);
            });`